src/modules/server.ts

```diff
@@ -1,20 +1,44 @@
 import Koa from 'koa'
+import mongoose from 'mongoose'
 import Router from 'koa-router'
 import getPort from 'get-port'
 import asyncRetry from 'async-retry'
 import hello from './handlers/hello'
+import info from './handlers/info'
 // eslint-disable-next-line no-unused-vars
 import { Server } from 'http'
-
+import dotenv from 'dotenv'
+dotenv.config()
 class WebServer {
   private app: Koa
   private router: Router
   private port: number
   private server: Server
+  private mongoUrl: string = `mongodb+srv://carolestrella:${process.env.PASSWORD}@cluster0.1inby.mongodb.net/noderest?retryWrites=true&w=majority`
 
   constructor () {
     this.app = new Koa()
     this.router = new Router()
+    this.mongoSetup()
+  }
+
+  private async mongoSetup (config = { useNewUrlParser: true, useUnifiedTopology: true }): Promise<void> {
+    try {
+      await mongoose.connect(this.mongoUrl, config)
+      console.log('Successfully connected to mongodb')
+    } catch (error) {
+      console.debug(`Failed to connect database. Reason: ${error.message}`)
+    }
+  }
+
+  public async dropDatabase () : Promise<void> {
+    await mongoose.connection.dropDatabase()
+    console.log('Successfully dropped database')
```

Comment:
If this method throws an error your code will still log `Successfully dropped database`, right ?

Reply:
Yes. I fixed it for a Try/Catch